lib/domainsTemplates.ts

```diff
@@ -135,16 +139,16 @@ export default class DomainTemplatesClient implements IDomainTemplatesClient {
 
     data.template = new DomainTemplateItem(response.body.template);
 
-    data.pages = response.body.paging;
+    data.pages = this.parsePageLinks(response, '?', 'p');
 
     return data;
   }
 
-  list(domain: string, query?: DomainTemplatesQuery): Promise<ListDomainTemplatesResult> {
-    return this.request.get(urljoin(this.baseRoute, domain, '/templates'), query)
-      .then(
-        (res: APIResponse) => this.parseList(res)
-      );
+  async list(domain: string, query?: DomainTemplatesQuery): Promise<ListDomainTemplatesResult> {
+    const { url, updatedQuery } = this.updateUrlAndQuery(urljoin(this.baseRoute, domain, '/templates'), query);
+
+    const response: ListDomainTemplatesAPIResponse = await this.request.get(url, updatedQuery);
+    return this.parseList(response);
```

Comment:
Here is the next place with boilerplate code

Reply:
Done